src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -817,20 +819,22 @@ public ThresholdingResult getAnomalyResultForEntity(
                 modelState.setModel(entityModel);
             }
 
-            // trainModelFromExistingSamples may be able to make models not null
-            if (entityModel.getRcf() == null || entityModel.getThreshold() == null) {
-                entityColdStarter.trainModelFromExistingSamples(modelState);
-            }
-
-            if (entityModel.getRcf() != null && entityModel.getThreshold() != null) {
-                return score(datapoint, modelId, modelState);
+            if (entityModel.getErcf().isPresent()) {
+                result = toResult(entityModel.getErcf().get().process(datapoint));
             } else {
-                entityModel.addSample(datapoint);
-                return new ThresholdingResult(0, 0, 0);
+                // trainModelFromExistingSamples may be able to make models not null
+                if (entityModel.getRcf() == null || entityModel.getThreshold() == null) {
+                    entityColdStarter.trainModelFromExistingSamples(modelState);
+                }
+
+                if (entityModel.getRcf() != null && entityModel.getThreshold() != null) {
+                    return score(datapoint, modelId, modelState);
+                } else {
+                    entityModel.addSample(datapoint);
+                }
```

Comment:
[Clarification question] Just to confirm, does invoking `entityColdStarter.trainModelFromExistingSamples(modelState)` synchronously change the result of `entityModel.getRcf()` and/or `entityModel.getThreshold()`? Otherwise the second if statement is redundant.

[Minor suggestion, non-blocking] Since this set of states is being checked twice (just in opposite cases), it would be more elegant and easier to read to add a method on `entityModel` that checks this exact condition. Then you could have:
```
if (entityModel.hasRcfOrThreshold()) { ... }
if (!entityModel.hasRcfOrThreshold() { ... } else { ... }
```

Reply:
1. Both should be set or unset. But it's fine and the code for old models will be removed soon as it is replaced by new ercf.
2. As above, this branch will be left as it is and soon be replaced by new logic based on ercf.